packages/babel-plugin/src/utils/normalize-props-usage.ts

```diff
@@ -108,6 +108,27 @@ const normalizeDestructuredString = (
   });
 };
 
+/**
+ * Create a member expression from a list of strings. For example, if
+ * `objectChain = ['a', 'b', 'c']`, the generated member expression will be
+ * `a.b.c`.
+ * @param objectChain List of strings
+ * @returns A member expression.
+ */
+const createNestedMemberExpression = (objectChain: string[]): t.Identifier | t.MemberExpression => {
+  if (objectChain.length === 1) return t.identifier(objectChain[0]);
+  if (objectChain.length > 1) {
+    const [init, last] = [
+      objectChain.slice(0, objectChain.length - 1),
+      objectChain[objectChain.length - 1],
+    ];
+    return t.memberExpression(createNestedMemberExpression(init), t.identifier(last));
+  }
+  throw new Error(
```

Comment:
Bit of a preference thing but it's a bit confusing to know when the error will throw. It might be explicit to throw an exception if the length === 0

Reply:
no worries, happy to add it for clarity